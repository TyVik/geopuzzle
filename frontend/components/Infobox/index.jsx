'use strict';
import React from "react";
import {Panel} from "react-bootstrap";
import localization from "../../localization";
import "./index.css";


class Infobox extends React.Component {
    componentWillMount() {
        this.setState({...this.state,
            collapse: JSON.parse(localStorage.getItem('infobox_collapse')) || false
        });
    }

    renderAttribute(name) {
        if (this.props[name]) {
            return <tr>
                <td>{localization[name]}</td>
                <td>{this.props[name]}</td>
            </tr>;
        } else {
            return null;
        }
    }

    toggleCollapse = () => {
        let value = !this.state.collapse;
        localStorage.setItem('infobox_collapse', value);
        this.setState({...this.state, collapse: value});
    };

    render() {
        if (!this.props.show) {
            return null;
        }

        let image = this.props.flag ? this.props.flag : this.props.coat_of_arms;
        return <div className="infobox">
            <div className="header">
                <button type="button" className="close" onClick={this.props.onClose}>
                    <span>&times;</span>
                </button>
                <div className="row_name">
                    {this.props.name} <sup><a href={this.props.wiki} target="_blank">wiki</a></sup>
                </div>
                <span
                    className={"glyphicon collapse-icon glyphicon-chevron-" + (this.state.collapse ? 'up' : 'down')}
                    onClick={this.toggleCollapse}>
                </span>
            </div>
            <Panel expanded={!this.state.collapse} onToggle={this.toggleCollapse}>
                <Panel.Collapse>
                    <Panel.Body>
                        <table>
                            <tbody>
                                {image &&
                                <tr>
                                    <td colSpan="2">
                                        <img src={image}/>
                                    </td>
                                </tr>
                                }
                                {this.props.capital &&
                                <tr>
                                    <td>{localization.capital}</td>
                                    <td><a href={this.props.capital.wiki} target="_blank">{this.props.capital.name}</a></td>
                                </tr>
                                }
                                {this.renderAttribute('area')}
                                {this.renderAttribute('population')}
                                {this.renderAttribute('currency')}
                            </tbody>
                        </table>
                    </Panel.Body>
                </Panel.Collapse>
            </Panel>
        </div>;
    }
}


export default Infobox;
