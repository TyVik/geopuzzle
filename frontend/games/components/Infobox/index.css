.infobox {
    z-index: 3;
    background: white;
    padding: 7px;
    border-radius: 5px;
    border: 1px solid black;
    width: 214px;
}

.infobox tr {
    line-height: 1.8em;
}

.infobox img {
    border: 1px solid black;
    width: 200px;
}

.header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
}

.header .collapse-icon {
    cursor: pointer;
}

.infobox .row_name {
    font-weight: bold;
    line-height: 1.2em;
    font-size: 1.25em;
    padding: 0 3px;
    text-align: center;
}

.infobox .panel {
    margin-bottom: 0;
    border: none;
}

.infobox .panel .collapse.in {
    margin-top: 5px;
}

.infobox .panel-body {
    padding: 0;
}
