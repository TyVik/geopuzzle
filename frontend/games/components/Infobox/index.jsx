'use strict';
import React from "react";
import {Collapse} from "react-bootstrap";
import {FormattedMessage as Msg} from "react-intl";

import "./index.css";


class Infobox extends React.Component {
  COLLAPSE_ID = 'infobox-collapse';

  constructor(props) {
    super(props);
    this.state = {collapse: JSON.parse(localStorage.getItem('infobox_collapse')) || false};
  }

  renderAttribute(name) {
    if (this.props[name]) {
      return <tr>
        <td><Msg id={name}/></td>
        <td>{this.props[name]}</td>
      </tr>;
    } else {
      return null;
    }
  }

  toggleCollapse = () => {
    let value = !this.state.collapse;
    localStorage.setItem('infobox_collapse', value);
    this.setState({...this.state, collapse: value});
  };

  render() {
    if (!this.props.show) {
      return null;
    }

    let image = this.props.flag ? this.props.flag : this.props.coat_of_arms;
    return <div className="infobox">
      <div className="infobox-header">
        <button type="button" className="close" onClick={this.props.onClose}>
          <span>&times;</span>
        </button>
        <div className="row_name">
          {this.props.name} <sup><a href={this.props.wiki} target="_blank">wiki</a></sup>
        </div>
        <i
          className={"fas fa-angle-" + (this.state.collapse ? 'up' : 'down')}
          onClick={this.toggleCollapse} aria-controls={this.COLLAPSE_ID} aria-expanded={!this.state.collapse}>
        </i>
      </div>
      <Collapse in={!this.state.collapse}>
        <div id={this.COLLAPSE_ID}>
          <table>
            <tbody>
              {image &&
                <tr>
                  <td colSpan="2">
                    <img src={image}/>
                  </td>
                </tr>}
              {this.props.capital &&
                <tr>
                  <td><Msg id="capital"/></td>
                  <td><a href={this.props.capital.wiki} target="_blank">{this.props.capital.name}</a></td>
                </tr>}
              {this.renderAttribute('area')}
              {this.renderAttribute('population')}
              {this.renderAttribute('currency')}
            </tbody>
          </table>
        </div>
      </Collapse>
    </div>;
  }
}


export default Infobox;
