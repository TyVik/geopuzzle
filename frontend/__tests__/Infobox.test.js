'use strict';
import React from 'react';
import {shallow} from 'enzyme';
import Infobox from '../games/components/Infobox';


describe('shallow <Infobox /> components', () => {
  let onClose = jest.fn();
  let props = {show: true, name: 'test', wiki: 'wiki', onClose: onClose, flag: 'flag', coat_of_arms: 'coat_of_arms', area: 'area', population: 'population', currency: 'currency', capital: {wiki: 'capital.wiki', name: 'capital.name'}};

  it('render hidden', () => {
    expect(shallow(<Infobox show={false}/>).html()).toBe(null);
  });

  it('check props', () => {
    expect(shallow(<Infobox {...props}/>)).toMatchSnapshot('infobox');
    expect(shallow(<Infobox show={true} coat_of_arms='coat_of_arms'/>)).toMatchSnapshot('infobox-empty');
  });

  it('click close', () => {
    let wrapper = shallow(<Infobox {...props}/>);
    wrapper.find('button.close').simulate('click');
    expect(onClose).toHaveBeenCalled();
  });

  it('click collapse', () => {
    let wrapper = shallow(<Infobox {...props}/>);
    wrapper.find('.infobox-header i').simulate('click');
    expect(localStorage.getItem('infobox_collapse')).toBe('true');
    expect(wrapper).toMatchSnapshot('infobox-collapse');
    wrapper.find('.infobox-header i').simulate('click');
    expect(localStorage.getItem('infobox_collapse')).toBe('false');
    expect(wrapper).toMatchSnapshot('infobox');
  });
});
