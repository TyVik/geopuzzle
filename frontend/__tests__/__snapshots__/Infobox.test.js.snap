// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow <Infobox /> components check props: infobox 1`] = `
<div
  className="infobox"
>
  <div
    className="infobox-header"
  >
    <button
      className="close"
      onClick={[MockFunction]}
      type="button"
    >
      <span>
        ×
      </span>
    </button>
    <div
      className="row_name"
    >
      test
       
      <sup>
        <a
          href="wiki"
          rel="noopener noreferrer"
          target="_blank"
        >
          wiki
        </a>
      </sup>
    </div>
    <i
      aria-controls="infobox-collapse"
      aria-expanded={true}
      className="fas fa-angle-down"
      onClick={[Function]}
    />
  </div>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={true}
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="infobox-collapse"
    >
      <table>
        <tbody>
          <tr>
            <td
              colSpan="2"
            >
              <img
                src="flag"
              />
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="capital"
                values={Object {}}
              />
            </td>
            <td>
              <a
                href="capital.wiki"
                rel="noopener noreferrer"
                target="_blank"
              >
                capital.name
              </a>
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="area"
                values={Object {}}
              />
            </td>
            <td>
              area
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="population"
                values={Object {}}
              />
            </td>
            <td>
              population
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="currency"
                values={Object {}}
              />
            </td>
            <td>
              currency
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </Collapse>
</div>
`;

exports[`shallow <Infobox /> components check props: infobox-empty 1`] = `
<div
  className="infobox"
>
  <div
    className="infobox-header"
  >
    <button
      className="close"
      type="button"
    >
      <span>
        ×
      </span>
    </button>
    <div
      className="row_name"
    >
       
      <sup>
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          wiki
        </a>
      </sup>
    </div>
    <i
      aria-controls="infobox-collapse"
      aria-expanded={true}
      className="fas fa-angle-down"
      onClick={[Function]}
    />
  </div>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={true}
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="infobox-collapse"
    >
      <table>
        <tbody>
          <tr>
            <td
              colSpan="2"
            >
              <img
                src="coat_of_arms"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </Collapse>
</div>
`;

exports[`shallow <Infobox /> components click collapse: infobox 1`] = `
<div
  className="infobox"
>
  <div
    className="infobox-header"
  >
    <button
      className="close"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      type="button"
    >
      <span>
        ×
      </span>
    </button>
    <div
      className="row_name"
    >
      test
       
      <sup>
        <a
          href="wiki"
          rel="noopener noreferrer"
          target="_blank"
        >
          wiki
        </a>
      </sup>
    </div>
    <i
      aria-controls="infobox-collapse"
      aria-expanded={true}
      className="fas fa-angle-down"
      onClick={[Function]}
    />
  </div>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={true}
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="infobox-collapse"
    >
      <table>
        <tbody>
          <tr>
            <td
              colSpan="2"
            >
              <img
                src="flag"
              />
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="capital"
                values={Object {}}
              />
            </td>
            <td>
              <a
                href="capital.wiki"
                rel="noopener noreferrer"
                target="_blank"
              >
                capital.name
              </a>
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="area"
                values={Object {}}
              />
            </td>
            <td>
              area
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="population"
                values={Object {}}
              />
            </td>
            <td>
              population
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="currency"
                values={Object {}}
              />
            </td>
            <td>
              currency
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </Collapse>
</div>
`;

exports[`shallow <Infobox /> components click collapse: infobox-collapse 1`] = `
<div
  className="infobox"
>
  <div
    className="infobox-header"
  >
    <button
      className="close"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      type="button"
    >
      <span>
        ×
      </span>
    </button>
    <div
      className="row_name"
    >
      test
       
      <sup>
        <a
          href="wiki"
          rel="noopener noreferrer"
          target="_blank"
        >
          wiki
        </a>
      </sup>
    </div>
    <i
      aria-controls="infobox-collapse"
      aria-expanded={false}
      className="fas fa-angle-up"
      onClick={[Function]}
    />
  </div>
  <Collapse
    appear={false}
    dimension="height"
    getDimensionValue={[Function]}
    in={false}
    mountOnEnter={false}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      id="infobox-collapse"
    >
      <table>
        <tbody>
          <tr>
            <td
              colSpan="2"
            >
              <img
                src="flag"
              />
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="capital"
                values={Object {}}
              />
            </td>
            <td>
              <a
                href="capital.wiki"
                rel="noopener noreferrer"
                target="_blank"
              >
                capital.name
              </a>
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="area"
                values={Object {}}
              />
            </td>
            <td>
              area
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="population"
                values={Object {}}
              />
            </td>
            <td>
              population
            </td>
          </tr>
          <tr>
            <td>
              <FormattedMessage
                id="currency"
                values={Object {}}
              />
            </td>
            <td>
              currency
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </Collapse>
</div>
`;
